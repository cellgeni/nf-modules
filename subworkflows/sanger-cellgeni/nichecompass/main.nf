include { NICHECOMPASS_TRAINING } from '../../../modules/sanger-cellgeni/imaging/nichecompass/training/main'
include { NICHECOMPASS_ANALYSIS } from '../../../modules/sanger-cellgeni/imaging/nichecompass/analysis/main'

workflow NICHECOMPASS {

    take:
    ch_h5ad // channel: [ val(meta), [ h5ad ] ]

    main:

    ch_versions = Channel.empty()

    NICHECOMPASS_TRAINING (ch_h5ad)
    ch_versions = ch_versions.mix(NICHECOMPASS_TRAINING.out.versions.first())

    NICHECOMPASS_ANALYSIS (NICHECOMPASS_TRAINING.out.nichecompass_model)
    ch_versions = ch_versions.mix(NICHECOMPASS_ANALYSIS.out.versions.first())

    emit:
    nichecompass_dir = NICHECOMPASS_ANALYSIS.out.nichecompass_dir  // channel: [ val(meta), [ nichecompass_dir ] ]
    notebook         = NICHECOMPASS_ANALYSIS.out.notebook          // channel: [ val(meta), [ notebook.ipynb ] ]
    versions         = ch_versions                                 // channel: [ versions.yml ]
}
