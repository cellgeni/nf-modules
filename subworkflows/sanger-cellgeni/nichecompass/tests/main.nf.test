nextflow_workflow {

    name "Test Subworkflow NICHECOMPASS"
    script "../main.nf"
    workflow "NICHECOMPASS"

    tag "subworkflows"
    tag "subworkflows_sangercellgeni"
    tag "subworkflows/nichecompass"
    tag "nichecompass/training"
    tag "nichecompass/analysis"


    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("nichecompass - mouse - 3") {

        when {
            params {
                spatial_nichecompasstraining_args = '--species mouse'
            }
            workflow {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '*.h5ad', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
