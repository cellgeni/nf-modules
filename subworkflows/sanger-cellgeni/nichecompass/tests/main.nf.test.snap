{
    "nichecompass - h5ad - sample_3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,cd3d04a786a5fff381063a22e9b028ec",
                                    "cell_types_latent_physical_space.svg:md5,dabaccb0d1101815eb7f9bee07695bb0",
                                    "enriched_gps_heatmap.svg:md5,864dfc2a29d5765ab212a79394b7b0df",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,75630887ebf510c974123301f313a01a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,b039085c240929b7aac63116704ea900",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,6cbda49d460005ae90d8bdf43f26c95a",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,e064938d834df203bf4cdcfd5807afe1",
                                    "res_0.2_niche_composition_batches.svg:md5,8c06a40c7e3299b45be236c5d291630c",
                                    "res_0.2_niche_composition_cell_types.svg:md5,a6d59a981c1dd072eb729b32b20abb9c",
                                    "res_0.2_niches_latent_physical_space.svg:md5,81d94aa94c2e85993cb22ef719ea4f25"
                                ],
                                [
                                    "attr.pkl:md5,fec73a217c93ed840df201ca2f4d397c",
                                    "model_params.pt:md5,bc1ab674d7523c609d8e0d8375942417",
                                    "trained.h5ad:md5,45010f2194f3196cb62ef9479144c676",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,668b1047cb3d4c8fa6e166ffa7d671f6",
                            "timestamp.txt:md5,695bd1fb92695e2a69bab6498100ea26",
                            "train.log:md5,edade2248fbca0c16f938acb3fa3116f"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250829_135211.ipynb:md5,ff91f9066005b4041763adf1105800b0"
                    ]
                ],
                "2": [
                    "versions.yml:md5,781e4dc74d5447ce054798dfb37e46ce",
                    "versions.yml:md5,94ac0b2217394474804b9942fafa8d99"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,cd3d04a786a5fff381063a22e9b028ec",
                                    "cell_types_latent_physical_space.svg:md5,dabaccb0d1101815eb7f9bee07695bb0",
                                    "enriched_gps_heatmap.svg:md5,864dfc2a29d5765ab212a79394b7b0df",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,75630887ebf510c974123301f313a01a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,b039085c240929b7aac63116704ea900",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,6cbda49d460005ae90d8bdf43f26c95a",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,e064938d834df203bf4cdcfd5807afe1",
                                    "res_0.2_niche_composition_batches.svg:md5,8c06a40c7e3299b45be236c5d291630c",
                                    "res_0.2_niche_composition_cell_types.svg:md5,a6d59a981c1dd072eb729b32b20abb9c",
                                    "res_0.2_niches_latent_physical_space.svg:md5,81d94aa94c2e85993cb22ef719ea4f25"
                                ],
                                [
                                    "attr.pkl:md5,fec73a217c93ed840df201ca2f4d397c",
                                    "model_params.pt:md5,bc1ab674d7523c609d8e0d8375942417",
                                    "trained.h5ad:md5,45010f2194f3196cb62ef9479144c676",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,668b1047cb3d4c8fa6e166ffa7d671f6",
                            "timestamp.txt:md5,695bd1fb92695e2a69bab6498100ea26",
                            "train.log:md5,edade2248fbca0c16f938acb3fa3116f"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250829_135211.ipynb:md5,ff91f9066005b4041763adf1105800b0"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,781e4dc74d5447ce054798dfb37e46ce",
                    "versions.yml:md5,94ac0b2217394474804b9942fafa8d99"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-29T13:56:45.033316278"
    },
    "nichecompass - mouse - 3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_20250830"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,183da62a69ca4fbb56b55d1c95944882",
                                    "cell_types_latent_physical_space.svg:md5,30b9eb382333aaf9cb630f46e5d4905d",
                                    "enriched_gps_heatmap.svg:md5,65662965b1a512a06ce7f664a265810e",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,a28eb5ea2ed611875581450cb1cd3423",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,728c1af5aa47a7779631bde29d5e495a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,8d414ee7406eff102b911b2a9e253f10",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,6b093719edb71ee539988e6d44c829b2",
                                    "res_0.2_niche_composition_batches.svg:md5,b7b78f150c5a4b6ccdb2ff86f92e267a",
                                    "res_0.2_niche_composition_cell_types.svg:md5,161a6ee967cd7568428757cbee821ef0",
                                    "res_0.2_niches_latent_physical_space.svg:md5,5c0f155d749e4bfb4ee9e63db061e175"
                                ]
                            ]
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_20250830"
                        },
                        "test_20250830_nichecompass_analysis.ipynb:md5,0c3131b7aef713f4065801ff6708ebf4"
                    ]
                ],
                "2": [
                    "versions.yml:md5,781e4dc74d5447ce054798dfb37e46ce",
                    "versions.yml:md5,94ac0b2217394474804b9942fafa8d99"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test_20250830"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,183da62a69ca4fbb56b55d1c95944882",
                                    "cell_types_latent_physical_space.svg:md5,30b9eb382333aaf9cb630f46e5d4905d",
                                    "enriched_gps_heatmap.svg:md5,65662965b1a512a06ce7f664a265810e",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,a28eb5ea2ed611875581450cb1cd3423",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,728c1af5aa47a7779631bde29d5e495a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,8d414ee7406eff102b911b2a9e253f10",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,6b093719edb71ee539988e6d44c829b2",
                                    "res_0.2_niche_composition_batches.svg:md5,b7b78f150c5a4b6ccdb2ff86f92e267a",
                                    "res_0.2_niche_composition_cell_types.svg:md5,161a6ee967cd7568428757cbee821ef0",
                                    "res_0.2_niches_latent_physical_space.svg:md5,5c0f155d749e4bfb4ee9e63db061e175"
                                ]
                            ]
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test_20250830"
                        },
                        "test_20250830_nichecompass_analysis.ipynb:md5,0c3131b7aef713f4065801ff6708ebf4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,781e4dc74d5447ce054798dfb37e46ce",
                    "versions.yml:md5,94ac0b2217394474804b9942fafa8d99"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-30T21:06:19.354397226"
    }
}